src/main/java/domain/Cards.java

```diff
@@ -0,0 +1,38 @@
+package domain;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Cards {
+
+    private static final int TOP_CARD = 0;
+    private static final int CARD_TOTAL_COUNT = 52;
+    private static final int FIRST_CARD_NUMBER = 0;
+
+    private final List<Card> cards;
+
+    public static Cards createAutoCards() {
+        return new Cards();
+    }
+
+    public Cards(final List<Card> cards) {
+        this.cards = cards;
+    }
+
+    private Cards() {
+        cards = new LinkedList<>();
+        init();
+        Collections.shuffle(cards);
+    }
```

Comment:
```suggestion
    public Cards(final List<Card> cards) {
        this.cards = cards;
    }

    public static Cards createAutoCards() {
        List<Card> cards = makeShuffledCards();
        return new Cards(cards);
    }

    private static List<Card> makeShuffledCards() {
        List<Card> cards = new LinkedList<>();

        for (int number = FIRST_CARD_NUMBER; number < CARD_TOTAL_COUNT; number++) {
            cards.add(Card.createWithMatchNumber(number));
        }

        Collections.shuffle(cards);

        return cards;
    }
```

이런 방식은 어떨까.. ㅎㅎ

더 좋은 방식이 있다면 다르게 바꿔도 되고~